bash/scripts/commit_description_parser.sh

```diff
@@ -16,42 +17,50 @@ parse-commit-description() {
 
   # Looks specifically for matches of '[rR]equires <repo>#<prNumber>',
   # e.g., "requires builder#123, Requires router#567"
-  reqs=$(echo "${description}" | grep -o "[Rr]equires \(deis\/\)\?[-a-z]*#[0-9]\{1,9\}" | grep -o "[-a-z]*#[0-9]\{1,9\}") || true
-
-  # split on whitespace into array of '<repo>#<prNumber>' values
-  reqsArray=(${reqs// / })
-
-  # for each '<repo>#<prNumber>' value in array
-  local i
-  for i in "${reqsArray[@]}"; do
-    # split on '#'
-    repoPRNumberArray=(${i//#/ })
-    repoName="${repoPRNumberArray[0]}"
-    prNumber="${repoPRNumberArray[1]}"
-
-    # curl GH api for needed commit sha, providing name and issue number
-    sha=$(get-most-recent-pr-commit "${repoName}" "${prNumber}")
-
-    if [[ ${sha} =~ ^[a-f0-9]{5,40}$ ]]; then
-      # echo '<uppercased repo>_SHA'=<prNumber> (with hyphens converted to underscores)
-      repoEnvVar=$(echo "${repoName//-/_}" | awk '{print toupper($0)}')_SHA
-      echo "${repoEnvVar}"="${sha}"
-    else
-      echo "Commit sha for PR #${prNumber} in repo '${repoName}' not found."
-      return 1
-    fi
-  done
+  reqs=$(echo "${description}" | grep -o "[Rr]equires \(deis\/\)\?[-a-z0-9]*#[0-9]\{1,9\}" | grep -o "[-a-z0-9]*#[0-9]\{1,9\}") || true
+
+  if [ "${reqs}" != "" ]; then
+    echo "Found Required PR(s)..." >&2
+    echo "${reqs}" >&2
+
+    # split on whitespace into array of '<repo>#<prNumber>' values
+    reqsArray=(${reqs// / })
+
+    # for each '<repo>#<prNumber>' value in array
+    local i
+    for i in "${reqsArray[@]}"; do
+      # split on '#'
+      repoPRNumberArray=(${i//#/ })
+      repoName="${repoPRNumberArray[0]}"
+      prNumber="${repoPRNumberArray[1]}"
+
+      # get all commits from prNumber in repoName, filtering for most recent, stripping quotes
+      sha=$(get-pr-commits "${repoName}" "${prNumber}" | tail -n1 | tr -d '"')
+
+      if [[ ${sha} =~ ^[a-f0-9]{5,40}$ ]]; then
+        echo "Using most recent commit: ${sha}" >&2
+        # echo '<uppercased repo>_SHA'=<prNumber> (with hyphens converted to underscores)
+        repoEnvVar=$(echo "${repoName//-/_}" | awk '{print toupper($0)}')_SHA
+        echo "${repoEnvVar}"="${sha}"
+      else
+        echo "Failure: Commit sha for PR #${prNumber} in repo '${repoName}' not found!" >&2
+        return 1
+      fi
+    done
+  fi
 }
 
-get-most-recent-pr-commit() {
+get-pr-commits() {
   repoName="${1}"
   prNumber="${2}"
 
-  curl \
+  resp=$(curl \
   -sSL \
   --user deis-admin:"${GITHUB_ACCESS_TOKEN}" \
-  "https://api.github.com/repos/deis/${repoName}/pulls/${prNumber}/commits" \
-  | docker run -i --rm kamermans/jq '.[].sha' | tail -n1
+  "https://api.github.com/repos/deis/${repoName}/pulls/${prNumber}/commits")
+
+  echo "${resp}" >&2
+  echo "${resp}" | docker run -i --rm kamermans/jq '.[].sha'
```

Comment:
@sgoings the return value from that `jq` command was actually a string, albeit multi-lined if more than one `sha` was found... therefore `last` wasn't the appropriate choice.